### PR TITLE
Added api.md 

### PR DESCRIPTION
**Pull request by shreyansjainn**
Added API documentation for an easier basic understanding of the gym API.

### CODE REVIEW

**Comment by jkterry1** [comment]
Upon cursory review, what you wrote looks decent. Please documentent all the action/observation spaces and the vector API though.

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+## Interacting with the Environment
+This example will run an instance of `CartPole-v0` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the classic [cart-pole](https://www.youtube.com/watch?v=J7E6_my3CHk&ab_channel=TylerStreeter) problem
+
+```python
+import gym
+env = gym.make('CartPole-v0')
+env.reset()
+for _ in range(1000): 
+	env.render() 
+	env.step(env.action_space.sample()) # take a random action
```
Comment (cclauss) [inline docs/api.md:19]:
```suggestion
	env.step(env.action_space.sample())  # take a random action 	
```
PEP8 Readability: Two spaces before an inline comment.
https://www.python.org/dev/peps/pep-0008/#inline-comments

**Comment by shreyansjainn** [comment]
Added All the spaces and Vector API Documentation

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+## Interacting with the Environment
+This example will run an instance of `CartPole-v0` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the classic [cart-pole](https://www.youtube.com/watch?v=J7E6_my3CHk&ab_channel=TylerStreeter) problem
+
+```python
+import gym
+env = gym.make('CartPole-v0')
+env.reset()
+for _ in range(1000): 
+	env.render() 
+	env.step(env.action_space.sample())  # take a random action
```
Comment (araffin) [inline docs/api.md:19]: dito

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+## Interacting with the Environment
+This example will run an instance of `CartPole-v0` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the classic [cart-pole](https://www.youtube.com/watch?v=J7E6_my3CHk&ab_channel=TylerStreeter) problem
+
+```python
+import gym
+env = gym.make('CartPole-v0')
+env.reset()
+for _ in range(1000): 
+	env.render()
```
Comment (araffin) [inline docs/api.md:18]:
```suggestion
	env.render() # by default `mode="human"`(GUI), you can pass `mode="rbg_array"` to retrieve an image instead
```

Comment (cclauss) [reply docs/api.md:18]: See https://github.com/openai/gym/pull/2303#discussion_r686492701

**Review by cclauss — APPROVED** [review]
LGTM

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+env = gym.make('CartPole-v0')
+```
+
+## Interacting with the Environment
+This example will run an instance of `CartPole-v0` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the classic [cart-pole](https://www.youtube.com/watch?v=J7E6_my3CHk&ab_channel=TylerStreeter) problem
+
+```python
+import gym
+env = gym.make('CartPole-v0')
+env.reset()
```
[the remark this answers is not in the document]
Comment (shreyansjainn) [reply docs/api.md:16]: Hey, thanks for the review, I've included a gif in the latest commit

[the remark this answers is not in the document]
Comment (araffin) [reply docs/api.md:16]: I meant the value it returns: `obs = env.reset()`

**Pull request merged**